test/parallel/test-listen-fd-cluster.js

```diff
@@ -7,63 +7,62 @@ var PORT = common.PORT;
 var spawn = require('child_process').spawn;
 var cluster = require('cluster');
 
-console.error('Cluster listen fd test', process.argv.slice(2));
+console.error('Cluster listen fd test', process.argv[2] || 'runner');
 
 if (common.isWindows) {
   console.log('1..0 # Skipped: This test is disabled on windows.');
   return;
 }
 
+// Process relationship is:
+//
+// parent: the test main script
+//   -> master: the cluster master
+//        -> worker: the cluster worker
 switch (process.argv[2]) {
   case 'master': return master();
   case 'worker': return worker();
   case 'parent': return parent();
-  default: return test();
 }
 
+var ok;
+
+process.on('exit', function() {
+  assert.ok(ok);
+});
+
 // spawn the parent, and listen for it to tell us the pid of the cluster.
 // WARNING: This is an example of listening on some arbitrary FD number
 // that has already been bound elsewhere in advance.  However, binding
 // server handles to stdio fd's is NOT a good or reliable way to do
 // concurrency in HTTP servers!  Use the cluster module, or if you want
 // a more low-level approach, use child process IPC manually.
-function test() {
-  var parent = spawn(process.execPath, [__filename, 'parent'], {
-    stdio: [ 0, 'pipe', 2 ]
-  });
-  var json = '';
-  parent.stdout.on('data', function(c) {
-    json += c.toString();
-    if (json.indexOf('\n') !== -1) next();
-  });
-  function next() {
-    console.error('output from parent = %s', json);
-    var cluster = JSON.parse(json);
-    // now make sure that we can request to the worker, then kill it.
-    http.get({
-      server: 'localhost',
-      port: PORT,
-      path: '/',
-    }).on('response', function(res) {
-      var s = '';
-      res.on('data', function(c) {
-        s += c.toString();
-      });
-      res.on('end', function() {
-        // kill the worker before we start doing asserts.
-        // it's really annoying when tests leave orphans!
-        parent.kill();
-        process.kill(cluster.master, 'SIGKILL');
-
+test(function(parent) {
+  // now make sure that we can request to the worker, then kill it.
+  http.get({
+    server: 'localhost',
+    port: PORT,
+    path: '/',
+  }).on('response', function(res) {
+    var s = '';
+    res.on('data', function(c) {
+      s += c.toString();
+    });
+    res.on('end', function() {
+      // kill the worker before we start doing asserts.
+      // it's really annoying when tests leave orphans!
+      parent.kill();
+      parent.on('exit', function() {
```

Comment:
I'd use `common.mustCall(function() {` here, then you can get rid of `ok` and the `process.on('exit', ...)` listener.